node/full_client_test.go

```diff
@@ -378,6 +381,66 @@ func TestGetCommit(t *testing.T) {
 	})
 }
 
+func TestCometBFTLightClientCompability(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+	mockApp, rpc := getRPC(t)
+	mockApp.On("FinalizeBlock", mock.Anything, mock.Anything).Return(finalizeBlockResponse)
+	mockApp.On("Commit", mock.Anything, mock.Anything).Return(&abci.ResponseCommit{}, nil)
+
+	// creat 3 consecutive signed blocks
+	block1, privKey := types.GetRandomBlockWithKey(1, 1)
+
+	block2 := types.GetRandomNextBlock(block1, privKey, []byte{}, 2)
+
+	block3 := types.GetRandomNextBlock(block2, privKey, []byte{}, 3)
+
+	blocks := []*types.Block{block1, block2, block3}
+
+	// start the node for testing
+	err := rpc.node.Start()
+	require.NoError(err)
+	defer func() {
+		assert.NoError(rpc.node.Stop())
+	}()
+
+	// save the 3 blocks
+	for _, b := range blocks {
+		err = rpc.node.Store.SaveBlock(b, &b.SignedHeader.Commit)
+		rpc.node.Store.SetHeight(b.Height())
+		require.NoError(err)
+	}
+
+	// Check if the block header provided by rpc.Commit() can be verified using tendermint light client
+	t.Run("checking rollkit ABCI header verifiability", func(t *testing.T) {
+		// trusted header to verify against
+		var trustedHeader cmtypes.SignedHeader
+		setTrustedHeader := false
+		// valset of single sequencer is constant
+		fixedValSet := block1.SignedHeader.Validators
+
+		// for each block (except block 1), verify it's ABCI header with previous block's ABCI header as trusted header
+		for _, b := range blocks {
+			h := int64(b.Height())
+			commit, err := rpc.Commit(context.Background(), &h)
+			require.NoError(err)
+			require.NotNil(commit)
+			require.NotNil(commit.SignedHeader)
+			assert.Equal(h, commit.Height)
+
+			// if there's no trusted header to verify against, set trusted header and skip verifying
+			if !setTrustedHeader {
+				trustedHeader = commit.SignedHeader
+				setTrustedHeader = true
+				continue
+			}
+
+			light.Verify(&trustedHeader, fixedValSet, &commit.SignedHeader, fixedValSet, 3*time.Hour, b.Time(), 10*time.Second, cmtmath.Fraction{Numerator: 2, Denominator: 1})
```

Comment:
The returned error should be checked here. After reviewing with @AdityaSripal we made some changes to make this pass. However changes in `utils.go` test helpers cause failures in other places in the repo. 

Here's a PR to the fork which has the changes for `light.Verify()` https://github.com/decentrio/rollkit/pull/1

Will require further work to ensure other existing tests are still passing.

Reply:
Should also add that `VotingPower` was required to be set greater than zero. We made this change in the testing utils but I'm unsure if its being assigned currently to the single validator https://github.com/decentrio/rollkit/pull/1#discussion_r1456422294

Reply:
@damiannolan voting power is always set to 1.
https://github.com/decentrio/rollkit/blob/6254cdb458b011af1b51a94cdba3196f34486a16/types/utils.go#L283-L290